### PR TITLE
Ensure DSL for Exacts

### PR DESCRIPTION
**Pull request by ustits**
- Added `ensure` dsl for `Exact`
- Fixed the directory naming (files were in directory `arrow.exact` by my mistake)

Decided to use `ensure` instead of `refine`, I think it will be better for users to have consistent naming throughout all Arrow modules. `ensure` accepts other `Exact` so now it must be easier to combine them.

The only problem is that I am not quite familiar with knit and probably have made something wrong, @nomisRev please fix me in that case 🙏 

Feel free to change this PR!

### CODE REVIEW

**Review by ILIYANGERMANOV — APPROVED** [review]
Excellent work @ustits! 👏 Can't wait to use it in my projects

I support the name change to `ensure` to keep it consistent with Arrow.

File: src/commonMain/kotlin/arrow/exact/ExactDsl.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+@ExactDsl
+public fun <E : Any, A, R> exactEither(construct: ExactScope<A, E>.() -> R): ExactEither<E, A, R> =
+  object : ExactEither<E, A, R> {
+    override fun from(value: A): Either<E, R> = either { construct(ExactScope(value, this)) }
+  }
+
+
+public class ExactScope<A, E : Any>(public val raw: A, private val raise: Raise<E>) : Raise<E> by raise {
+
+  public fun <B> ensure(exact: ExactEither<E, A, B>): B {
```
Comment (ILIYANGERMANOV) [inline src/commonMain/kotlin/arrow/exact/ExactDsl.kt:24]: Can you annotate it the `ExactDsl` dsl marker so Android Studio and IntelliJ can highlight it?

**Review by ILIYANGERMANOV — APPROVED** [review]

**Review by nomisRev — APPROVED** [review]
This looks super awesome @ustits 🙌 Great work, I only have two small nits, so I am going to approve. We can update or change them later if we want to ☺️

Feel free to merge 🥳

File: src/commonMain/kotlin/arrow/exact/ExactDsl.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import arrow.core.raise.Raise
+import arrow.core.raise.either
+
+@DslMarker public annotation class ExactDsl
+
+@ExactDsl
+public fun <A, R> exact(construct: ExactScope<A, ExactError>.() -> R): Exact<A, R> =
+  object : Exact<A, R> {
+    override fun from(value: A): Either<ExactError, R> = either { construct(ExactScope(value, this)) }
+  }
```
[the remark this answers is not in the document]
Comment (ustits) [reply src/commonMain/kotlin/arrow/exact/ExactDsl.kt]: You are right, also will change `ExactEither`

**Comment by nomisRev** [comment]
Thank you @ustits 🙌

**Pull request merged**